examples/efficientdet/efficientdet_building_blocks.py

```diff
@@ -0,0 +1,321 @@
+                                        target_height,
+                                        target_width))
+
+        return feat
+
+
+class FPNCells(tf.keras.layers.Layer):
+    """Set of FPN Cells."""
+
+    def __init__(self, config, name='fpn_cells'):
+        super().__init__(name=name)
+        self.config = config
+        self.fpn_config = get_fpn_configuration(config['fpn_name'],
+                                                config['min_level'],
+                                                config['max_level'],
+                                                config['fpn_weight_method']
+                                                )
+        self.cells = [FPNCell(self.config, name='cell_%d' % repeats)
+                      for repeats in range(self.config['fpn_cell_repeats'])]
+
+    def call(self, feats, training):
+        for cell in self.cells:
+            cell_feats = cell(feats, training)
+            min_level = self.config['min_level']
+            max_level = self.config['max_level']
+
+            feats = []
+            for level in range(min_level, max_level + 1):
+                for i, fnode in enumerate(reversed(self.fpn_config['nodes'])):
+                    if fnode['feat_level'] == level:
+                        feats.append(cell_feats[-1 - i])
+                        break
+        return feats
+
+
+class FPNCell(tf.keras.layers.Layer):
+    """A single FPN cell."""
+    def __init__(self, config, name='fpn_cell'):
+        super().__init__(name=name)
+        self.config = config
+        self.fpn_config = get_fpn_configuration(config['fpn_name'],
+                                                config['min_level'],
+                                                config['max_level'],
+                                                config['fpn_weight_method']
+                                                )
+        self.fnodes = []
+        for i, fnode_cfg in enumerate(self.fpn_config['nodes']):
+            fnode = FNode(
+                fnode_cfg['feat_level'] - self.config['min_level'],
+                fnode_cfg['inputs_offsets'],
+                config['fpn_num_filters'],
+                config['apply_bn_for_resampling'],
+                config['conv_after_downsample'],
+                config['conv_bn_act_pattern'],
+                config['separable_conv'],
+                config['act_type'],
+                weight_method=self.config['fpn_weight_method'],
+                data_format=self.config['data_format'],
+                name='fnode%d' % i)
+            self.fnodes.append(fnode)
+
+    def call(self, feats, training):
+
+        for fnode in self.fnodes:
+            feats = fnode(feats, training)
+        return feats
+
+
+def add_n(nodes):
+    """A customized add_n to add up a list of tensors."""
+    with tf.name_scope('add_n'):
+        new_node = nodes[0]
+        for n in nodes[1:]:
+            new_node = new_node + n
+        return new_node
```

Comment:
is this summing up tensors? If so maybe changing the function name would be better. Also I would prefer to not use any name_scope since it might make it harder later to change in case API changes or in case we change dependencies.